functions/auth.js

```diff
@@ -19,3 +30,164 @@ exports.passwordsMatch = (submittedPassword, actualPassword) =>
   submittedPassword.trim() === actualPassword.trim() ||
   lowercaseFirstChar(submittedPassword.trim()) ===
     lowercaseFirstChar(actualPassword.trim());
+
+/**
+ * Redirect the user to the authentication consent screen. The 'state' cookie is set for later state verification.
+ *
+ * @see https://github.com/lelylan/simple-oauth2/blob/3.x/API.md#authorizeurlauthorizeoptions--string
+ */
+// TODO: rename this authorize or similar?
+exports.connectI4AOAuth = functions.https.onRequest(async (req, res) => {
+  const { venueId } = req.query;
+
+  if (!AUTH_ORIGIN) {
+    throw new HttpsError("internal", "auth.origin is not configured properly");
+  }
+
+  if (!checkIfValidVenueId(venueId)) {
+    throw new HttpsError("invalid-argument", "venueId is not a valid venue id");
+  }
+
+  const authConfig = await fetchAuthConfig(venueId);
+
+  const authClient = createOAuth2Client(authConfig);
+
+  // TODO: configure this in cloud config and/or firestore or similar
```

Comment:
```suggestion
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1523/commits/28638e8502fe6cfd5e7b11b43493cb15271c4326